test/extensions/filters/http/composite/composite_filter_integration_test.cc

```diff
@@ -66,15 +149,78 @@ TEST_P(CompositeFilterIntegrationTest, TestBasic) {
   }
 
   {
-    const Http::TestRequestHeaderMapImpl request_headers = {{":method", "GET"},
-                                                            {":path", "/somepath"},
-                                                            {":scheme", "http"},
-                                                            {"match-header", "match"},
-                                                            {":authority", "blah"}};
-    auto response = codec_client_->makeRequestWithBody(request_headers, 1024);
+    auto response = codec_client_->makeRequestWithBody(match_request_headers_, 1024);
     ASSERT_TRUE(response->waitForEndStream());
     EXPECT_THAT(response->headers(), Http::HttpStatusIs("403"));
   }
 }
 
+// Verifies function of the per-route config in the ExtensionWithMatcher class.
+TEST_P(CompositeFilterIntegrationTest, TestPerRoute) {
+  config_helper_.prependFilter(absl::StrFormat(filter_config_template, "composite"));
+  addPerRouteResponseCodeFilter("composite", "/somepath", 401, false);
+  initialize();
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+
+  auto response = codec_client_->makeRequestWithBody(match_request_headers_, 1024);
+  ASSERT_TRUE(response->waitForEndStream());
+  EXPECT_THAT(response->headers(), Http::HttpStatusIs("401"));
+}
+
+// Verifies function of the per-route config in the ExtensionWithMatcher class with multiple routes.
+TEST_P(CompositeFilterIntegrationTest, TestMultiplePerRoute) {
+  config_helper_.prependFilter(absl::StrFormat(filter_config_template, "composite"));
+  addPerRouteResponseCodeFilter("composite", "/somepath", 401, false);
+  addPerRouteResponseCodeFilter("composite", "/otherpath", 402, true);
+  initialize();
+
+  {
+    codec_client_ = makeHttpConnection(lookupPort("http"));
+    auto response = codec_client_->makeRequestWithBody(match_request_headers_, 1024);
+    ASSERT_TRUE(response->waitForEndStream());
+    EXPECT_THAT(response->headers(), Http::HttpStatusIs("401"));
+  }
+
+  cleanupUpstreamAndDownstream();
+
+  {
+    codec_client_ = makeHttpConnection(lookupPort("http"));
+    const Http::TestRequestHeaderMapImpl custom_request_headers_ = {{":method", "POST"},
+                                                                    {":path", "/otherpath"},
+                                                                    {":scheme", "http"},
+                                                                    {"match-header", "match"},
+                                                                    {":authority", "blah"}};
+    auto response = codec_client_->makeRequestWithBody(custom_request_headers_, 1024);
+    ASSERT_TRUE(response->waitForEndStream());
+    EXPECT_THAT(response->headers(), Http::HttpStatusIs("402"));
+  }
+}
+
+// Tests a filter chain with routes.
+TEST_P(CompositeFilterIntegrationTest, TestPerRouteFilterChain) {
```

Comment:
This test seems to not use either of the overwritten configurations.
Even when both filters "composite_2" and "composite" get per_route configs the default config is still the one that sets the 403 status.
Even though the body filter is applied.

Reply:
This seems to happen regardless of the order of the filters.

Reply:
Note: `set_reponse_code` filter sets the response code by sending local reply, which will terminate the filter iteration to next filter. So, in order to test multiple `SetReponse` filters, you will want the first setResponse filter return `continue` rather than `localReply` 

It can be achieved this by specifying the `prefix` field in config like below
```
typed_config:
   "@type": type.googleapis.com/test.integration.filters.SetResponseCodeFilterConfig
     prefix: "skipLocalReplyInFirstFilterSoThatWeCanContinue"
     code: 466
```
More details can be found [here](https://github.com/envoyproxy/envoy/blob/5e7a4b5f44f398a42897f743a8665ce92fc563a3/test/integration/filters/set_response_code_filter.cc#LL35C76-L35C83)

Reply:
Perfect, that's exactly what I needed.